anime_search_engine.py

```diff
@@ -0,0 +1,95 @@
+from whoosh.index import create_in, open_dir, exists_in
+from whoosh.fields import *
+from whoosh.analysis import StemmingAnalyzer
+from whoosh.qparser import QueryParser, OrGroup, AndGroup
+from whoosh import scoring
+
+from bs4 import BeautifulSoup
+
+import os, pickle
+
+class SearchEngine(object):
+
+	def __init__(self):
+		self.schema = Schema(title=TEXT(stored=True), url = ID(stored=True), content=TEXT(analyzer=StemmingAnalyzer()))
+		self.ix = self.__get_indexer("./indexdir")
+		self.limit = 10 # Number of results displayed
+		self.conj = True
+		self.size = self.ix.doc_count()
+		self.page_rank = self.__unpickle("./page_rank.dat")
+		assert self.size == len(self.page_rank), "the index and page rank don't match"
+
+	# Returns an existing or new indexer 
+	def __get_indexer(self, path):
+		if not os.path.exists(path): os.mkdir(path)
+		if not exists_in(path): return self.__create_index(path)
+		else: return open_dir(path)
+
+	# Get url map from path file
+	def __unpickle(self, path):
+		data = None;
+		with open(path,"rb") as f:
+			data = pickle.load(f)
+		return data
+
+	# Returns a new indexer with a Whoosh Index
+	def __create_index(self, path):
+		ix = create_in(path, self.schema)
+		writer = ix.writer(limitmb=1024, procs=4, multisegment=True)
+		urls = self.__unpickle("./sample/url_map.dat")
```

Comment:
Again, having class variables or function parameters for file names would probably be a good idea.